sycl/unittests/Extensions/CommandGraph.cpp

```diff
@@ -1244,10 +1244,101 @@ TEST_F(CommandGraphTest, MakeEdgeErrors) {
   CheckGraphStructure();
 }
 
-TEST_F(CommandGraphTest, EnqueueBarrierExceptionCheck) {
-  testEnqueueBarrier<OperationPath::Explicit>();
-  testEnqueueBarrier<OperationPath::RecordReplay>();
-  testEnqueueBarrier<OperationPath::Shortcut>();
+TEST_F(CommandGraphTest, ExplicitBarrierException) {
+
+  std::error_code ExceptionCode = make_error_code(sycl::errc::success);
+  try {
+    auto Barrier =
+        Graph.add([&](sycl::handler &cgh) { cgh.ext_oneapi_barrier(); });
+  } catch (exception &Exception) {
+    ExceptionCode = Exception.code();
+  }
+  ASSERT_EQ(ExceptionCode, sycl::errc::invalid);
+}
+
+TEST_F(CommandGraphTest, EnqueueBarrier) {
```

Comment:
A test that uses more than one barrier in a graph I think would also be good.

Reply:
I added a unitest that enqueues two barriers and also added a test-e2e using barrier (forgot in the initial commit)